src/adb.js

```diff
@@ -487,6 +488,220 @@ class Adb {
         );
     });
   }
+
+  // AW : return the file size of a complete folder
+  getFileSize(file) {
+    var _this = this;
+    return new Promise(function(resolve, reject) {
+      _this.getAdbState().then(stdout => {
+        if (stdout == 1) {
+          // Recovery
+          _this
+            .shell("du -shk " + file)
+            .then(stdout => {
+              stdout = parseFloat(stdout);
+              _this.log("FileSize is " + stdout + " Ko");
+              resolve(stdout);
+            })
+            .catch(reject);
+        } else if (stdout == 2) {
+          // device (type : used, size)
+          _this
+            .shell("df -hBK " + file + " --output=used|tail -n1")
+            .then(stdout => {
+              stdout = parseFloat(stdout);
+              _this.log("FileSize is " + stdout + " Ko");
+              resolve(stdout);
+            })
+            .catch(reject);
+        }
+      });
+    }).catch(e => {
+      _this.log(e);
+      reject;
+    });
+  }
+
+  // AW : Return the status of the device (0 : bootloader, 1 : recovery, 2 : device ...)
+  getAdbState() {
```

Comment:
Rename function to `getState()`. We generally try to stay close to the original names.